bitcoind-rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindConfig.scala

```diff
@@ -321,4 +340,22 @@ object BitcoindConfig {
     .toPath()
     .resolve("bitcoin.conf")
     .toFile
+
+  /**
+    * Writes the config to the data directory within it, it it doesn't
+    * exist. Returns the written file.
+    */
+  def writeConfigToFile(config: BitcoindConfig, datadir: File): Path = {
+
+    val confStr = config.lines.mkString("\n")
+
+    Files.createDirectories(datadir.toPath)
+    val confFile = datadir.toPath.resolve("bitcoin.conf")
+
+    if (!Files.exists(confFile)) {
```

Comment:
I think this screws with `withOption`. Updating the config doesn't result in a updated file, which doesn't seem to be what we want

Reply:
I guess the alternative here is risking that we overrwrite a bitcoin.conf with important information in it. I'm not sure what the best way to balance this is? We could check if it is the `DEFAULT_DATADIR`, but this still leaves the possibility of users losing money because their rpc credentials were in a non traditional datadir location.